valueformatting_test.go

```diff
@@ -0,0 +1,613 @@
+}
+
+func TestBinaryValueFormaterINT32(t *testing.T) {
+	checkBinaryValueFormatter(
+		t, "int32", 16, "[66051 67438087 -1 -100]", binary.BigEndian)
+}
+
+func TestBinaryValueFormaterINT64(t *testing.T) {
+	checkBinaryValueFormatter(
+		t, "int64", 16, "[283686952306183 -100]", binary.BigEndian)
+}
+
+func TestBinaryValueFormaterUINT8(t *testing.T) {
+	checkBinaryValueFormatter(
+		t, "uint8", 16, "[0 1 2 3 4 5 6 7 255 255 255 255 255 255 255 156]",
+		binary.BigEndian)
+}
+
+func TestBinaryValueFormaterUINT16(t *testing.T) {
+	checkBinaryValueFormatter(
+		t, "uint16", 16, "[1 515 1029 1543 65535 65535 65535 65436]",
+		binary.BigEndian)
+}
+
+func TestBinaryValueFormaterUINT32(t *testing.T) {
+	checkBinaryValueFormatter(
+		t, "uint32", 16, "[66051 67438087 4294967295 4294967196]", binary.BigEndian)
+}
+
+func TestBinaryValueFormaterUINT64(t *testing.T) {
+	checkBinaryValueFormatter(
+		t, "uint64", 16, "[283686952306183 18446744073709551516]", binary.BigEndian)
+}
+
+func TestBinaryValueFormaterFLOAT32(t *testing.T) {
+	checkBinaryValueFormatter(
+		t, "float32", 16, "[9.2557e-41 1.5636842e-36 NaN NaN]", binary.BigEndian)
+}
+
+func TestBinaryValueFormaterFLOAT64(t *testing.T) {
+	checkBinaryValueFormatter(
+		t, "float64", 16, "[1.40159977307889e-309 NaN]", binary.BigEndian)
+}
+
+func TestValueFormattingBinaryFormatter(t *testing.T) {
+	formatting := newValueFormatting()
+	var formatter = formatting.binaryFormatter("BigEndian", "int32")
+	s, err := formatter(TestBinaryFormaterTestData)
+	want := "[66051 67438087 -1 -100]"
+
+	if err != nil {
+		t.Errorf("Error when creating formatter: %v", err)
+	}
+	if s != want {
+		t.Errorf("Binary value formatted incorrectly: wanted %s, got %s",
+			want, s)
+	}
+
+	formatter = formatting.binaryFormatter("LittleEndian", "int32")
+	s, err = formatter(TestBinaryFormaterTestData)
+	want = "[50462976 117835012 -1 -1660944385]"
+
+	if err != nil {
+		t.Errorf("Error when creating formatter: %v", err)
+	}
+
+	if s != want {
+		t.Errorf("Binary value formatted incorrectly: wanted %s, got %s",
+			want, s)
+	}
+}
+
+func TestValueFormattingPBFormatter(t *testing.T) {
+	formatting := newValueFormatting()
+	formatting.settings.ProtocolBufferDefinitions = append(
+		formatting.settings.ProtocolBufferDefinitions,
+		filepath.Join("testdata", "addressbook.proto"))
+	err := formatting.setupPBMessages()
+	if err != nil {
+		t.Errorf("Error creating protobuf formatter: %v", err)
+	}
+
+	formatter := formatting.pbFormatter("person")
+	in, err := ioutil.ReadFile(filepath.Join("testdata", "person.bin"))
+	if err != nil {
+		t.Errorf("Error reading testdata: %v", err)
+	}
+
+	got, err := formatter(in)
+	want := `name: "Jim"
+id: 42
+email: "jim@example.com"
+phones: <
+  number: "555-1212"
+  type: HOME
+>`
+
+	if err != nil {
+		t.Errorf("Error creating protobuf formatter: %v", err)
+	}
+
+	if got != want {
+		t.Errorf("Protobuf not formatted correctly: wanted %s; got %s",
+			want, got)
+	}
+
+	/*
```

Comment:
More commented code here.

Reply:
I was able to fix this! Yay!

Reply:
I was able to address the bug in the underlying proto library by adding some more nil checks.